exec/Chainweb/Gaps.hs

```diff
@@ -215,32 +186,27 @@ withDroppedIndexes pool logger action = do
     fmap fst $ generalBracket (dropIndexes pool indexInfos) release (const action)
   where
     release _ = \case
-      ExitCaseSuccess _ -> dedupeTables pool logger
+      ExitCaseSuccess _ -> return () -- dedupeTables pool logger
       _ -> return ()
 
-getBlockGaps :: Env -> IO (M.Map Int64 [(Int64,Int64)])
-getBlockGaps env = P.withResource (_env_dbConnPool env) $ \c -> do
-    runBeamPostgresDebug (liftIO . logg Debug) c $ do
-      let toMap = M.fromListWith (<>) . map (\(cid,a,b) -> (cid,[(a,b)]))
-      foundGaps <- fmap toMap $ runSelectReturningList $ selectWith $ do
-        foundGaps <- selecting $
-          withWindow_ (\b -> frame_ (partitionBy_ (_block_chainId b)) (orderPartitionBy_ $ asc_ $ _block_height b) noBounds_)
-                      (\b w -> (_block_chainId b, _block_height b, lead_ (_block_height b) (val_ (1 :: Int64)) `over_` w))
-                      (all_ $ _cddb_blocks database)
-        pure $ orderBy_ (\(cid,a,_) -> (desc_ a, asc_ cid)) $ do
-          res@(_,a,b) <- reuse foundGaps
-          guard_ ((b - a) >. val_ 1)
-          pure res
-      minHeights' <- chainMinHeights
-      let minHeights =
-            M.intersectionWith
-              maybeAppendGenesis
-              minHeights'
-              $ fmap toInt64 $ M.mapKeys toInt64 genesisInfo
-      unless (M.null minHeights) (liftIO $ logg Debug $ fromString $ "minHeight: " <> show minHeights)
-      pure $ if M.null foundGaps
-        then M.mapMaybe (fmap pure) minHeights
-        else M.intersectionWith addStart minHeights foundGaps
+getBlockGaps :: Env -> M.Map Int64 (Maybe Int64) -> IO (M.Map Int64 [(Int64,Int64)])
+getBlockGaps env existingMinHeights = withDbDebug env Debug $ do
+    let toMap = M.fromListWith (<>) . map (\(cid,a,b) -> (cid,[(a,b)]))
+    foundGaps <- fmap toMap $ runSelectReturningList $ selectWith $ do
+      foundGaps <- selecting $
+        withWindow_ (\b -> frame_ (partitionBy_ (_block_chainId b)) (orderPartitionBy_ $ asc_ $ _block_height b) noBounds_)
+                    (\b w -> (_block_chainId b, _block_height b, lead_ (_block_height b) (val_ (1 :: Int64)) `over_` w))
+                    (all_ $ _cddb_blocks database)
+      pure $ orderBy_ (\(cid,a,_) -> (desc_ a, asc_ cid)) $ do
+        res@(_,a,b) <- reuse foundGaps
+        guard_ ((b - a) >. val_ 1)
+        pure res
+    let minHeights = M.intersectionWith maybeAppendGenesis existingMinHeights
```

Comment:
Don't need to calculate minHeights again because it was run earlier and can now be passed in.